src/views/InteractiveMap/Escher.vue

```diff
@@ -139,6 +164,81 @@ export default Vue.extend({
     },
     showDiffFVAScore() {
       return this.card ? this.card.showDiffFVAScore : false;
+    },
+    enzymeUsage() {
+      // If this is an enzyme-constrained model, calculate the enzyme usage for
+      // each pseudo-reaction.
+      if (
+        !this.card ||
+        !this.card.fluxes ||
+        !this.model ||
+        !this.model.model_serialized ||
+        !this.model.ec_model
+      ) {
+        return null;
+      }
+
+      // Each pseudoreaction always has a single gene, so store the enzyme usage
+      // value per gene.
+      const enzymeUsagePerGene = {};
+      // If the upper bound is smaller than this value, we can assume it was
+      // constrained by real-world observations. Anything larger is ignored,
+      // mostly to save computational cycles below.
+      const highestConveivableEnzymeUsage = 4;
+      this.model.model_serialized.reactions
+        .filter(
+          rxn => rxn.id.startsWith("prot_") && rxn.id.endsWith("_exchange")
+        )
+        .filter(rxn => rxn.upper_bound < highestConveivableEnzymeUsage)
+        // Ignore reactions with no positive flux - they would cause division by
+        // zero.
+        .filter(rxn => rxn.upper_bound !== 0)
+        .forEach(pseudoReaction => {
+          enzymeUsagePerGene[pseudoReaction.gene_reaction_rule] =
+            this.card.fluxes[pseudoReaction.id] / pseudoReaction.upper_bound;
+        });
+
+      // Now map those usages on to all reactions that are catalyzed by any of
+      // those genes.
+      const enzymeUsages = {};
+      this.model.model_serialized.reactions
+        .filter(
+          rxn => !(rxn.id.startsWith("prot_") && rxn.id.endsWith("_exchange"))
+        )
+        .forEach(rxn => {
+          rxn.gene_reaction_rule.split(/\W+/).forEach(gene => {
+            if (gene in enzymeUsagePerGene) {
+              // It's a match! Set the enzyme usage.
+              if (rxn.id in enzymeUsages) {
+                // Another gene is already recorded for this reaction; use the
+                // highest value.
+                enzymeUsages[rxn.id] = Math.max(
+                  enzymeUsages[rxn.id],
+                  enzymeUsagePerGene[gene]
+                );
+              } else {
+                enzymeUsages[rxn.id] = enzymeUsagePerGene[gene];
+              }
+            }
+          });
+        });
+
+      return enzymeUsages;
+    },
+    enzymeUsageMapped() {
+      // Map the ecModel-specific reaction identifiers back to the original
+      // identifiers, so that they are correctly recognized in the Escher map.
+      const mappedReactions = {};
+      Object.keys(this.enzymeUsage).forEach(id => {
+        let newId = id;
+        if (id.startsWith("arm_")) {
+          newId = id.slice(4);
+        } else if (id.endsWith("No1")) {
+          newId = id.slice(0, -3);
```

Comment:
@kvikshaug note that a `No1` reaction should only be chosen if there is no associated `arm` reaction in the whole network (as it is done in `ecModelFluxes()`), otherwise you are over-writing usage of the `arm` reaction with the value in `No1` (which might be 0 if `No2`, `No3`, etc. are chosen by the simulations instead). So the function should be something like:

```suggestion
          const rootId = id.slice(0, -3);
          if (!("arm_" + rootId in this.enzymeUsage)) {
            newId = rootId;
          }
```

Note that compared to `ecModelFluxes()` we can disregard the reversible reactions (`_REV`), as those will just contain redundant information in terms of enzyme usage (the GPR rule of the backward and forward reaction are equivalent).